src/NewsService.Business/EditNewsCommand.cs

```diff
@@ -1,50 +1,85 @@
-using System;
 
 namespace LT.DigitalOffice.NewsService.Business
 {
-    public class EditNewsCommand : IEditNewsCommand
+  public class EditNewsCommand : IEditNewsCommand
+  {
+    private readonly INewsRepository _repository;
+    private readonly IPatchNewsMapper _mapper;
+    private readonly IEditNewsValidator _validator;
+    private readonly IAccessValidator _accessValidator;
+    private readonly IHttpContextAccessor _httpContextAccessor;
+
+    public EditNewsCommand(
+      INewsRepository repository,
+      IPatchNewsMapper mapper,
+      IEditNewsValidator validator,
+      IAccessValidator accessValidator,
+      IHttpContextAccessor httpContextAccessor)
     {
+      _repository = repository;
+      _mapper = mapper;
+      _validator = validator;
+      _accessValidator = accessValidator;
+      _httpContextAccessor = httpContextAccessor;
+    }
 
-        private readonly INewsRepository _repository;
-        private readonly IPatchNewsMapper _mapper;
-        private readonly IEditNewsValidator _validator;
-        private readonly IAccessValidator _accessValidator;
+    public OperationResultResponse<bool> Execute(Guid newsId, JsonPatchDocument<EditNewsRequest> request)
+    {
+      if (!(_accessValidator.HasRights(Rights.AddEditRemoveNews)))
+      {
+        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.Forbidden;
 
-        public EditNewsCommand(
-            INewsRepository repository,
-            IPatchNewsMapper mapper,
-            IEditNewsValidator validator,
-            IAccessValidator accessValidator)
+        return new OperationResultResponse<bool>
         {
-            _repository = repository;
-            _mapper = mapper;
-            _validator = validator;
-            _accessValidator = accessValidator;
-        }
+          Status = OperationResultStatusType.Failed,
+          Errors = new() { "Not enough rights." }
+        };
+      }
+
+      if (!_validator.ValidateCustom(request, out List<string> errors))
+      {
+        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
 
-        public bool Execute(Guid newsId, JsonPatchDocument<EditNewsRequest> request)
+        return new OperationResultResponse<bool>
         {
-            if (!(_accessValidator.IsAdmin() ||
-                  _accessValidator.HasRights(Rights.AddEditRemoveNews)))
-            {
-                throw new ForbiddenException("Not enough rights.");
-            }
+          Status = OperationResultStatusType.Failed,
+          Errors = errors
+        };
+      }
+
+      OperationResultResponse<bool> response = new();
+
+      JsonPatchDocument<DbNews> dbRequest = _mapper.Map(request);
+
+      response.Body = _repository.Edit(newsId, dbRequest);
 
-            _validator.ValidateAndThrowCustom(request);
+      if (!response.Body)
+      {
+        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
```

Comment:
why it is bad request, request was correct and we cannot apply patch to model, it is just fail, where user mistake?

Reply:
done

Reply:
strange, in the studio changed, but in git it remained